e2e/rook-test-framework/clients/k8s_rook_block.go

```diff
@@ -0,0 +1,138 @@
+//Output  - k8s delete pvc operation output and/or error
+func (rb *k8sRookBlock) Block_Delete(name string) (string, error) {
+	cmdArgs := []string{name}
+	out, err, status := rb.transportClient.Delete(cmdArgs, nil)
+	if status == 0 {
+		return out, nil
+	} else {
+		return err, errors.New("Unable to delete Block")
+	}
+}
+
+// Function to list all the blocks created/being managed by rook
+//Returns a ouput  for rook cli for block list
+func (rb *k8sRookBlock) Block_List() (string, error) {
+	out, err, status := rb.transportClient.Execute(listCmd, nil)
+	if status == 0 {
+		return out, nil
+	} else {
+		return err, errors.New("Unable to list all blocks")
+	}
+}
+
+// Function to map a Block using Rook
+// Input paramaters -
+//name - path to a yaml file that creates a pod  - pod should be defined to use a pvc that was created earlier
+//mountpath - not used in this impl since mountpath is defined in the pod definition
+//Output  - k8s create pod operation output and/or error
+func (rb *k8sRookBlock) Block_Map(name string, mountpath string) (string, error) {
+	cmdArgs := []string{name}
+	out, err, status := rb.transportClient.Create(cmdArgs, nil)
+	if status == 0 {
+		return out, nil
+	} else {
+		return err, errors.New("Unable to Map block")
+	}
+
+}
+
+// Function to write  data to block created by rook ,i.e. write data to a pod that is using a pvc
+// Input paramaters -
+//name - path to a yaml file that creates a pod  - pod should be defined to use a pvc that was created earlier
+//mountpath - folder on the pod were data is supposed to be written(should match the mountpath described in the pod definition)
+//data - data to be written
+//filename - file where data is written to
+//namespace - optional param - namespace of the pod
+//Output  - k8s exec pod operation output and/or error
+func (rb *k8sRookBlock) Block_Write(name string, mountpath string, data string, filename string, namespace string) (string, error) {
+	wt := "echo \"" + data + "\">" + mountpath + "/" + filename
+	wideDataToBlockPod[2] = wt
+	option := []string{name}
+	if namespace != "" {
+		option = append(option, namespace)
+	}
+	out, err, status := rb.transportClient.Execute(wideDataToBlockPod, option)
+	if status == 0 {
+		return out, nil
+	} else {
+		return err, errors.New("Unable to write data to pod")
+	}
+}
+
+// Function to write  read from block created by rook ,i.e. Read data from a pod that is using a pvc
+// Input paramaters -
+//name - path to a yaml file that creates a pod  - pod should be defined to use a pvc that was created earlier
+//mountpath - folder on the pod were data is supposed to be written(should match the mountpath described in the pod definition)
+//filename - file to be read
+//namespace - optional param - namespace of the pod
+//Output  - k8s exec pod operation output and/or error
+func (rb *k8sRookBlock) Block_Read(name string, mountpath string, filename string, namespace string) (string, error) {
+	rd := mountpath + "/" + filename
```

Comment:
using the `filepath.Join()` function is a more preferred way to concatenate file paths.  It's platform specific too.  I think there are probably a lot of these file path concats that could be done with `filepath.Join()` instead.